index.rst

```diff
@@ -46,14 +46,314 @@
+  * `durability` - providing fault-tolerant storage for both real-time state data as well as persistent settings
+  * `remote control and monitoring SOAP service` - providing remote web-based access using the SOAP protocol from various Vortex OpenSplice tools
+  * `dbms service` - providing a connection between the real-time and the enterprise domain by bridging data from DDS to DBMS and vice versa
+
+By default only the `networking` and `durability` services are configured.
+
+Furthermore, OpenSplice DDS can be configured to run in two modes of operation, :ref:`single process <fig-single-process>` or :ref:`shared memory <fig-shared-memory>`.
+When running in single process mode, the configured DDS services will run in the same process as the application.
+That means, alongside the resources needed by the application itself to run (computing, memory allocation, etc) there will also be additional resources needed by the DDS services themselves.
+In addition, there is also the question of how those additional resources are handled by the application and the programming language itself.
+For instance, Python does not easily support multiprocessing, so DDS services will be competing for the same computing resources as those needed by the application to run properly.
+
+.. figure:: /_static/SingleProcessFig.png
+   :name: fig-single-process
+   :target: ../_images/SingleProcessFig.png
+   :alt: DDS single process mode
+
+   DDS architecture when running in single process mode.
+   In the left hand panel, a diagram showing the internal architecture of a single application.
+   The "DDS Application Process" represents the software developed on top of SAL/SalObj to communicate over DDS and execute some operation.
+   In the case of Rubin Observatory system this would be a CSC, salkafka producer, user notebook server, etc.
+   In the right hand panel, a high level overview of a group of applications (CSCs, salkafka producers, etc).
+   Each application runs their own DDS services alongside their processes.
+
+As one can imagine, this architecture will not scale well for large scale systems.
+If the DDS communication grows large enough (in terms of number of topics and published frequency) the processing requirements for the DDS services alone may start to interfere with the application and vice-versa.
+To overcome this problem, OpenSplice DDS can be configured to run in shared memory mode.
+
+When running in shared memory mode, the user configures and runs a single daemon per node that is dedicated to running the OpenSplice services.
+Applications running in the same node can then be configured to attach to the daemon using system shared memory.
+In this case, several applications share the same DDS services.
+The computing resources needed by the DDS services are separated from those of the application and will not interfere with each other (assuming the node has enough resources available).
```

Comment:
So the bit in the bracket raises a red flag. 
How hard is the resource balancing, what happens when resources get tight etc...

Reply:
When you have multiple processes running on a server, the OS ends up dealing with the resource balancing... I guess, we would have similar issues if a node run out of resources.